inc/authldap.class.php

```diff
@@ -1900,11 +1922,16 @@ static function getAllUsers(array $options, &$results, &$limitexceeded) {
                         && !$limitexceeded) {
                // Only manage deleted user if ALL (because of entity visibility in delegated mode)
 
-               //If user is marked as coming from LDAP, but is not present in it anymore
-               if (!$user['is_deleted']
-                   && ($user['auths_id'] == $options['authldaps_id'])) {
-                  User::manageDeletedUserInLdap($user['id']);
-                  $results[self::USER_DELETED_LDAP] ++;
+               if ($user['auths_id'] === $options['authldaps_id']) {
+                  if (!$user['is_deleted']) {
+                     // User is marked as coming from LDAP, but is not present in it anymore
+                     User::manageDeletedUserInLdap($user['id']);
+                     $results[self::USER_DELETED_LDAP] ++;
+                  } else {
```

Comment:
Unless I am wrong, at this point, user has not been found in LDAP directory (`$userfound` is equal to false). So we should not restore user here.